packages/@aws-cdk/aws-vpcv2-alpha/lib/route.ts

```diff
@@ -0,0 +1,788 @@
+    });
+
+    this.routerId = this.resource.attrId;
+  }
+}
+
+/**
+ * Creates a transit gateway
+ * @resource AWS::EC2::TransitGateway
+ */
+// export class TransitGateway extends Resource implements IRouteTarget {
+//   /**
+//    * The type of router used in the route.
+//    */
+//   readonly routerType: RouterType;
+
+//   /**
+//    * The ID of the route target.
+//    */
+//   readonly routerId: string;
+
+//   /**
+//    * The transit gateway CFN resource.
+//    */
+//   public readonly resource: CfnTransitGateway;
+
+//   constructor(scope: Construct, id: string/*, props: TransitGatewayProps*/) {
+//     super(scope, id);
+
+//     this.routerType = RouterType.TRANSIT_GATEWAY;
+
+//     this.resource = new CfnTransitGateway(this, 'TGW', {});
+
+//     this.routerId = this.resource.attrId;
+//   }
+// }
+
+/**
+ * Creates a request for a VPC peering connection between two VPCs: a requester VPC that
+ * you own and an accepter VPC with which to create the connection.
+ * @resource AWS::EC2::VPCPeeringConnection
+ */
+export class VpcPeeringConnection extends Resource implements IRouteTarget {
+  /**
+   * The type of router used in the route.
+   */
+  readonly routerType: RouterType;
+
+  /**
+   * The ID of the route target.
+   */
+  readonly routerId: string;
+
+  /**
+   * The VPC peering connection CFN resource.
+   */
+  public readonly resource: CfnVPCPeeringConnection;
+
+  constructor(scope: Construct, id: string, props: VpcPeeringConnectionProps) {
+    super(scope, id);
+
+    this.routerType = RouterType.VPC_PEERING_CONNECTION;
+
+    this.resource = new CfnVPCPeeringConnection(this, 'VPCPeerConnection', {
+      peerRoleArn: props.peerRole,
+      ...props,
+    });
+
+    this.routerId = this.resource.attrId;
+  }
+}
+
+/**
+ * The type of endpoint or gateway being targeted by the route.
+ */
+export interface IRouteTargetType {
+  /**
+   * The gateway route target. This is used for targets such as
+   * egress-only internet gateway or VPC peering connection.
+   * @default none
+   */
+  readonly gateway?: IRouteTarget;
+
+  /**
+   * The endpoint route target. This is used for targets such as
+   * VPC endpoints.
+   * @default none
+   */
+  readonly endpoint?: IVpcEndpoint;
+}
+
+/**
+ * The gateway or endpoint targeted by the route.
+ */
+export class RouteTargetType {
+  /**
+   * The gateway route target. This is used for targets such as
+   * egress-only internet gateway or VPC peering connection.
+   * @default none
+   */
+  readonly gateway?: IRouteTarget;
+
+  /**
+   * The endpoint route target. This is used for targets such as
+   * VPC endpoints.
+   * @default none
+   */
+  readonly endpoint?: IVpcEndpoint;
+
+  constructor(props: IRouteTargetType) {
+    if (props.gateway && props.endpoint) {
+      throw new Error('Cannot specify both gateway and endpoint');
+    };
+    if (props.gateway) {
+      this.gateway = props.gateway;
+    } else if (props.endpoint) {
+      this.endpoint = props.endpoint;
```

Comment:
can you remove the if/else check. just assign what ever you have in props without checking if it null or not